solr/core/src/test/org/apache/solr/request/TestWriterPerf.java

```diff
@@ -179,18 +175,9 @@ void doPerf(String writerName, SolrQueryRequest req, int encIter, int decIter) t
     System.gc();
     RTimer timer = new RTimer();
     for (int i = 0; i < encIter; i++) {
-      if (w instanceof BinaryQueryResponseWriter binWriter) {
-        out = new ByteArrayOutputStream();
-        binWriter.write(out, req, rsp);
-        out.close();
-      } else {
-        out = new ByteArrayOutputStream();
-        // to be fair, from my previous tests, much of the performance will be sucked up
-        // by java's UTF-8 encoding/decoding, not the actual writing
-        Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8);
```

Comment:
I dropped this branch because in normal use we'll write to an OutputStream, not a Writer.